webapp/src/Controller/API/ProblemController.php

```diff
@@ -371,6 +372,43 @@ public function singleAction(Request $request, string $id): Response
         return $this->renderData($request, $object);
     }
 
+    /**
+     * Get the statement for given problem for this contest.
+     * @throws NonUniqueResultException
+     * @Rest\Get("/{id}/statement")
+     * @OA\Response(
+     *     response="200",
+     *     description="Returns the given problem statement for this contest",
+     *     @OA\MediaType(mediaType="application/pdf")
+     * )
+     * @OA\Parameter(ref="#/components/parameters/id")
+     * @OA\Parameter(ref="#/components/parameters/strict")
+     */
+    public function statementAction(Request $request, string $id): Response
+    {
+        $queryBuilder = $this->getQueryBuilder($request)
+            ->addSelect('partial p.{probid,problemtext}')
+            ->setParameter('id', $id)
+            ->andWhere(sprintf('%s = :id', $this->getIdField()));
+
+        // Get the one result; we know it's only one since we filter on ID
+        $contestProblemData = $queryBuilder->getQuery()->getOneOrNullResult();
+
+        if (empty($contestProblemData)) {
+            throw new NotFoundHttpException(sprintf('Problem with ID \'%s\' not found', $id));
+        }
+
+        // The result contains the contest problem as well as the test data count. So get only the problem
```

Comment:
nit: make it more explicit that we don't want to leak any data here

Reply:
@vmcj i uh ik you offered to merge this. Could you also fix this nit? And the conflict

Reply:
Yes, will do it later today/tomorrow.

Reply:
Since I made the proposal I went ahead and pushed a nit fix. Will wait for CI to run and then merge.